doc/design/inference.md

```diff
@@ -0,0 +1,105 @@
+# Design Doc: Inferencer
+
+In fluid, a nueral network is represented as a protobuf message [ProgramDesc](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md), the python wrapper of which is `Program`.
+Given a `ProgramDesc`, it can be run on any execution environment.
+In fluid, we call the execution environment `Runtime`, which includes `Place`, `Scope` and `Executor`.
+
+## Representation of the Inference Network
+
+In python, an inference network is defined as:
+
+```python
+image = fluid.layers.data(name='x', shape=[784], dtype='float32')
+predict = fluid.layers.fc(input=image,
+                          size=10,
+                          act='softmax')
+```
+
+After training for serval passes, the parameters can be saved use `fluid.io.save_inference_model`, which will save the binary proto string of the network at the same time.
+```python
+fluid.io.save_inference_model(
+                "./inference_model/", ["x"], [predict],
+                exe)
+```
+
+The saved model contains everything of the inference network, including all operators and variables. Thus, the `inference_program` should be initilized by the model file or a pre-loaded buffer.

```

Comment:
"everything of the" => everything required by the

Reply:
Done.